File: No_3211/Generate Binary Strings Without Adjacent Zeros.cpp
class Solution {
public:
    void generator(vector<string> &ret, string &s, int n) {
        int strLen = s.size();
        if (strLen > 1 && s[strLen-1] == '0' && s[strLen-2] == '0') {
            return;
        }

        if (n == 0) {
            ret.push_back(s);
            return;
        }

        s += '0';
        generator(ret, s, n-1);
        s.pop_back();

        s += '1';
        generator(ret, s, n-1);
        s.pop_back();

        return;
    }

    vector<string> validStrings(int n) {
        vector<string> ret;
        string s = "";
        generator(ret, s, n);
        return ret;
    }
};

File: No_3211/Generate Binary Strings Without Adjacent Zeros.py
class Solution:
    def validStrings(self, n: int) -> List[str]:
        def generator(ret: list[str], s: string, n: int):
            if len(s) > 1 and s[-1] == '0' and s[-2] == '0':
                return

            if n == 0:
                ret.append(s)
                return

            s += '0'
            generator(ret, s, n-1)
            s = s[:-1]

            s += '1'
            generator(ret, s, n-1)
            s = s[:-1]

        ret = []
        generator(ret, "", n)
        return ret
        